File: Backend/NodeJs-server/source/common/entities.ts
import { AppError, UserType } from "./enums";
import { Request } from "express";
export interface entityWithId {
    id: number;
}

export interface campaign extends entityWithId {
    hashtag: string;
    landingPage: string;
    nonProfitOrganizationName?: string;
}

export interface campaignWitnProducts extends campaign {
    productTitle: string;
    businessOwnerName: string;
    productQty: number;

}

// id - campaignId
export interface productsForCampaign extends entityWithId {
    productId: number;
    productTitle: string;
    productPrice: string;
    productQty: string;
    companyName: string;
}

export interface user extends entityWithId {
    userTypeId: number;
    login: string;
    password: string;
}

export interface socialActivist extends user {
    twitterHandle: string;
    email: string;
    address: string;
    phoneNumber: string;
}

export interface businessOwner extends user {
    twitterHandle: string;
    name: string;
    email: string;
}

export interface nonProfitOrganization extends user {
    name: string;
    email: string;
    website: string;
}

export interface systemError {
    key: AppError
    code: number;
    message: string;
}

export interface authenticationToken {
    userId: number;
    userTypeId: UserType;
}

export interface AuthenticatedRequest extends Request, authenticationToken { }

File: Frontend/Blazor/BlazorWASM/Entities/Entities.cs
namespace BlazorWASM.Entities
{
    public class UserCredentials
    {
        public string Login { get; set; } = "";
        public string Password { get; set; } = "";
    }

    public class AuthToken
    {
        public string? Token { get; set; }
    }

    public class Campaign
    {
        public int Id { get; set; }
        public string Hashtag { get; set; }
        public string LandingPage { get; set; }
        public string NonProfitOrganizationName { get; set; }
    }

    public class NonProfitUser
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Email { get; set; }
        public string? Website { get; set; }
    }

    public class BusinessOwner
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Email { get; set; }
        public string? TwitterHandle { get; set; }
    }

    public class SocialActivist
    {
        public int Id { get; set; }
        public string? TwitterHandle { get; set; }
        public string? Email { get; set; }
        public string? Address { get; set; }
        public string? PhoneNumber { get; set; }
    }

    public class Tweet
    {
        public string? User { get; set; }
        public DateTime PublishedOn { get; set; }
        public string? Link { get; set; }
        public int Retweets { get; set; }
        public string? Body { get; set; }
    }
}
